packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useState } from 'react';
+import {
+  DataFrame,
+  FieldConfig,
+  FieldType,
+  formattedValueToString,
+  getFieldColorModeForField,
+  getTimeField,
+  systemDateFormats,
+} from '@grafana/data';
+import { timeFormatToTemplate } from '../uPlot/utils';
+import { alignAndSortDataFramesByFieldName } from './utils';
+import { Area, Axis, Line, Point, Scale, SeriesGeometry } from '../uPlot/geometries';
+import { UPlotChart } from '../uPlot/Plot';
+import { GraphCustomFieldConfig, PlotProps } from '../uPlot/types';
+import { useTheme } from '../../themes';
+
+const timeStampsConfig = [
+  [3600 * 24 * 365, '{YYYY}', 7, '{YYYY}'],
+  [3600 * 24 * 28, `{${timeFormatToTemplate(systemDateFormats.interval.month)}`, 7, '{MMM}\n{YYYY}'],
+  [
+    3600 * 24,
+    `{${timeFormatToTemplate(systemDateFormats.interval.day)}`,
+    7,
+    `${timeFormatToTemplate(systemDateFormats.interval.day)}\n${timeFormatToTemplate(systemDateFormats.interval.year)}`,
+  ],
+  [
+    3600,
+    `{${timeFormatToTemplate(systemDateFormats.interval.minute)}`,
+    4,
+    `${timeFormatToTemplate(systemDateFormats.interval.minute)}\n${timeFormatToTemplate(
+      systemDateFormats.interval.day
+    )}`,
+  ],
+  [
+    60,
+    `{${timeFormatToTemplate(systemDateFormats.interval.second)}`,
+    4,
+    `${timeFormatToTemplate(systemDateFormats.interval.second)}\n${timeFormatToTemplate(
+      systemDateFormats.interval.day
+    )}`,
+  ],
+  [
+    1,
+    `:{ss}`,
+    2,
+    `:{ss}\n${timeFormatToTemplate(systemDateFormats.interval.day)} ${timeFormatToTemplate(
+      systemDateFormats.interval.minute
+    )}`,
+  ],
+  [
+    1e-3,
+    ':{ss}.{fff}',
+    2,
+    `:{ss}.{fff}\n${timeFormatToTemplate(systemDateFormats.interval.day)} ${timeFormatToTemplate(
+      systemDateFormats.interval.minute
+    )}`,
+  ],
+];
+
+const defaultFormatter = (v: any) => (v == null ? '-' : v.toFixed(1));
+
+const TIME_FIELD_NAME = 'Time';
+
+interface GraphNGProps extends Omit<PlotProps, 'data'> {
+  data: DataFrame[];
+}
+
+export const GraphNG: React.FC<GraphNGProps> = ({ data, children, ...plotProps }) => {
```

Comment:
This component will replace the current Graph component in grafana/ui when we replace the current graph in Explore